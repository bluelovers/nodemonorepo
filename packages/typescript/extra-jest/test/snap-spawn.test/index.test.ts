import {spawnSync} from 'child_process'
import snapSpawn from '../../lib/snap-spawn'

describe('passing functions', () => {
  it('executes echo', () => snapSpawn(
    (argv, options) => {
      expect({
        arguments: {argv, options}
      }).toMatchSnapshot()

      return spawnSync('echo', argv, options)
    },
    ['Hello, World!!'],
    {encoding: 'utf8'}
  )())

  it('executes node', () => snapSpawn(
    (argv, options) => {
      expect({
        arguments: {argv, options}
      }).toMatchSnapshot()

      return spawnSync('node', argv, options)
    },
    [],
    {
      input: 'console.log({abc: 123, def: 456})'
    }
  )())
})

describe('passing command', () => {
  it(
    'executes echo',
    snapSpawn.withCommand('echo', ['Hello, World!!'])
  )

  it(
    'executes node',
    snapSpawn.withCommand('node', [], {input: 'console.log({abc: 123, def: 456})'})
  )
})
