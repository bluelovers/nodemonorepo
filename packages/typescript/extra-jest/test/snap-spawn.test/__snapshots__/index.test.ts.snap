// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passing command executes echo 1`] = `
"
error: null
signal: null
status: 0
stderr: ((EMPTY))
stdout: |+

  Hello, World!!


"
`;

exports[`passing command executes node 1`] = `
"
error: null
signal: null
status: 0
stderr: ((EMPTY))
stdout: |+

  { abc: 123, def: 456 }


"
`;

exports[`passing functions executes echo 1`] = `
Object {
  "arguments": Object {
    "argv": Array [
      "Hello, World!!",
    ],
    "options": Object {
      "encoding": "utf8",
    },
  },
}
`;

exports[`passing functions executes echo 2`] = `
"
error: null
signal: null
status: 0
stderr: ((EMPTY))
stdout: |+

  Hello, World!!


"
`;

exports[`passing functions executes node 1`] = `
Object {
  "arguments": Object {
    "argv": Array [],
    "options": Object {
      "input": "console.log({abc: 123, def: 456})",
    },
  },
}
`;

exports[`passing functions executes node 2`] = `
"
error: null
signal: null
status: 0
stderr: ((EMPTY))
stdout: |+

  { abc: 123, def: 456 }


"
`;
